opflexagent/utils/bridge_managers/vpp_manager.py

```diff
@@ -0,0 +1,143 @@
+
+    def __init__(self):
+        super(VppManager, self).__init__()
+
+    def initialize(self, host, opflex_conf, agent_state):
+        self.int_br_device_count = 0
+        try:
+            vpp_config = cfg.CONF.VPP
+        except cfg.NoSuchOptError:
+            raise
+        agent_state['vhostuser_socket_dir'] = vpp_config.vhostuser_socket_dir
+        return self, agent_state
+
+    def get_local_ip(self):
+        return None
+
+    def check_bridge_status(self):
+        vapi = VPPApi(LOG, 'gbp-agent')
+        version = vapi.get_version()
+        if (version['retval'] == 0) and version['version']:
+            return constants.OVS_NORMAL
+        return constants.OVS_DEAD
+
+    def setup_integration_bridge(self):
+        """Override parent setup integration bridge.
+        VPP renderer handles the bridge creation.
+        Nothing to do here
+        """
+        pass
+
+    def scan_ports(self, registered_ports, updated_ports=None):
+        cur_ports = self.get_vif_port_set()
+        self.int_br_device_count = len(cur_ports)
+        port_info = {'current': cur_ports}
+        updated_ports = updated_ports or set()
+        if updated_ports:
+            # Some updated ports might have been removed in the
+            # meanwhile, and therefore should not be processed.
+            # In this case the updated port won't be found among
+            # current ports.
+            updated_ports &= cur_ports
+            if updated_ports:
+                port_info['updated'] = updated_ports
+
+        if cur_ports == registered_ports:
+            # No added or removed ports to set, just return here
+            return port_info
+
+        port_info['added'] = cur_ports - registered_ports
+        # Remove all the known ports not found on the integration bridge
+        port_info['removed'] = registered_ports - cur_ports
+        return port_info
+
+    def get_port_vif_name(self, port_id, bridge=None):
+        bridge = bridge or self.int_br
+        ports = bridge.get_vifs_by_ids([port_id])
+        if ports:
+            return ports[port_id].port_name
+
+    def get_patch_port_pair_names(self, port_id):
+        return ("", "")
+
+    def _get_patch_peer_attrs(self, peer_name, port_id, port_mac=None):
+        external_ids = {}
+        if port_mac:
+            external_ids['attached-mac'] = port_mac
+        if port_id:
+            external_ids['iface-id'] = port_id
+        attrs = [('type', 'patch'), ('options', {'peer': peer_name})]
+        if external_ids:
+            attrs.append(('external_ids', external_ids))
+        return attrs
+
+    # VPP renderer handles the patch ports internally
+    def add_patch_ports(self, port_ids, attached_macs=None):
+        pass
+
+    def delete_patch_ports(self, port_ids):
+        pass
+
+    def process_deleted_port(self, port_id):
+        port = self.get_vif_port_by_id(port_id)
+        # move to dead VLAN so deleted ports no
+        # longer have access to the network
```

Comment:
Is this comment valid? It seems like port_dead is a noop

Reply:
Yes I think this should just be a noop( as it is for ovs).. will remove the dead code